From cad0a4a809447fbcb6559a5e52f4414d3bc26d8c Mon Sep 17 00:00:00 2001
From: Jagadeesh Kona <jagadeesh.kona@oss.qualcomm.com>
Date: Sat, 1 Aug 2026 00:42:45 +0530
Subject: [PATCH 12832/15333] dt-bindings: clock: qcom: Add Maili Graphics
 Clock Controllers

Maili gpucc is a derivative of Hawi gpucc with pll0 configuration
change, and hence reuses the Hawi gpucc driver with Maili fixup.
Maili gxclkctl is the same as Kaanapali gxclkctl and falls back to
it. Add the compatibles for the graphics clock controllers on the
Qualcomm Maili SoC.

Signed-off-by: Jagadeesh Kona <jagadeesh.kona@oss.qualcomm.com>
Acked-by: Krzysztof Kozlowski <krzysztof.kozlowski@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260801-maili_gpucc-v1-1-9d8c37bded69@oss.qualcomm.com
Signed-off-by: Bjorn Andersson <andersson@kernel.org>
---
 Documentation/devicetree/bindings/clock/qcom,hawi-gpucc.yaml  | 4 +++-
 .../devicetree/bindings/clock/qcom,kaanapali-gxclkctl.yaml    | 1 +
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/clock/qcom,hawi-gpucc.yaml b/Documentation/devicetree/bindings/clock/qcom,hawi-gpucc.yaml
index 3e40bc3c061d..b7179d53166f 100644
--- a/Documentation/devicetree/bindings/clock/qcom,hawi-gpucc.yaml
+++ b/Documentation/devicetree/bindings/clock/qcom,hawi-gpucc.yaml
@@ -17,7 +17,9 @@ description: |
 
 properties:
   compatible:
-    const: qcom,hawi-gpucc
+    enum:
+      - qcom,hawi-gpucc
+      - qcom,maili-gpucc
 
   clocks:
     items:
diff --git a/Documentation/devicetree/bindings/clock/qcom,kaanapali-gxclkctl.yaml b/Documentation/devicetree/bindings/clock/qcom,kaanapali-gxclkctl.yaml
index 28b653f6b64d..94f84d5294c6 100644
--- a/Documentation/devicetree/bindings/clock/qcom,kaanapali-gxclkctl.yaml
+++ b/Documentation/devicetree/bindings/clock/qcom,kaanapali-gxclkctl.yaml
@@ -28,6 +28,7 @@ properties:
       - items:
           - enum:
               - qcom,hawi-gxclkctl
+              - qcom,maili-gxclkctl
           - const: qcom,kaanapali-gxclkctl
 
   power-domains:
-- 
2.55.0

